mono/mini/debugger-agent.c

```diff
@@ -1112,6 +1112,10 @@ mono_debugger_disconnect ()
 	// loop already handles the debugger client disconnection properly,
 	// so calling transport_close2 method is enough since the method
 	// only closes the debugger client socket.
+
+	//restart debugger thread now that we've forcefully disconnected any clients
+	mono_atomic_cas_i32(&inited, 0, 1);
+	finish_agent_init(FALSE);
 	transport_close2();
```

Comment:
This doesn't look right, it should be calling `stop_debugger_thread` I think. I'll pull this branch and do a little testing in a bit to see if I can confirm my suspicions.

Reply:
Ok. I will wait for your response. Thanks.